### DISCUSSION TITLE
ESP32 :  fails to build the mcuboot, zephyr v3.1.0 rc2,  sdk 0.14.2

### DISCUSSION THREAD

**Discussion by sam131208** [zephyrproject-rtos/zephyr#46055] [General]
**Describe the bug**
Fails to build the mcuboot with sdk0.14.2.

**Logs and console output**
B250M-D3H:~/zephyrproject/bootloader/mcuboot/boot/espressif$ cmake -DCMAKE_TOOLCHAIN_FILE=tools/toolchain-esp32.cmake -DMCUBOOT_TARGET=esp32 -DMCUBOOT_FLASH_PORT=/dev/ttyUSB0 -B build -GNinja
CMake Error at /usr/share/cmake-3.23/Modules/CMakeDetermineSystem.cmake:130 (message):
  Could not find toolchain file: tools/toolchain-esp32.cmake
Call Stack (most recent call first):
  CMakeLists.txt:11 (project)

CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!

**Environment (please complete the following information):**
 - OS: Ubuntu
 - Toolchain:   Zephyr SDK  0.14.2
 - Zephyr Version 3.1 rc2

**Comment by sylvioalves** [comment]
@sam131208 It is necessary to follow all the steps to allow mcuboot building:  
https://github.com/mcu-tools/mcuboot/blob/main/docs/readme-espressif.md#installing-requirements-and-dependencies  

Would you retry?

Thanks.

**Comment by sam131208** [comment]
@sylvioalves Yes, i do it.
  I used another computer to build the whole zephyr, and she can compile it. But "ninja --build build/" also got an error, see the log below.
  However, the same results are still obtained on the original computer. Maybe there is some problem with the computer, I will rebuild the whole zephyr.

Logs and console output("ninja --build build/" ):
B250M-D3H:~/zephyrproject/bootloader/mcuboot/boot/espressif$ ninja --build build/
ninja: unrecognized option '--build'
usage: ninja [options] [targets...]

if targets are unspecified, builds the 'default' target (see manual).

options:
  --version      print ninja version ("1.10.0")
  -v, --verbose  show all command lines while building

  -C DIR   change to DIR before doing anything else
  -f FILE  specify input build file [default=build.ninja]

  -j N     run N jobs in parallel (0 means infinity) [default=6 on this system]
  -k N     keep going until N jobs fail (0 means infinity) [default=1]
  -l N     do not start new jobs if the load average is greater than N
  -n       dry run (don't run commands but act like they succeeded)

  -d MODE  enable debugging (use '-d list' to list modes)
  -t TOOL  run a subtool (use '-t list' to list subtools)
    terminates toplevel options; further flags are passed to the tool
  -w FLAG  adjust warnings (use '-w list' to list warnings)

**Reply by sylvioalves** [reply]
@sam131208 The `ninja --build build/` is wrong. I think there is a PR for that already. Replace `ninja` to `cmake`: `cmake --build build/`.
